# Use proper function (gdk_device_get_state) to get next motion hint
#
# Copyright (C) 2006 - 2007 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license, either version 2
# of the License, or (at your option) any later version.
#
diff -ru gtkhtml-3.13.91/src/gtkhtml.c gtkhtml-3.9.1.mod/src/gtkhtml.c
--- gtkhtml-3.13.91/src/gtkhtml.c	2006-12-04 12:43:55.000000000 +0200
+++ gtkhtml-3.13.91.mod/src/gtkhtml.c	2006-12-05 09:09:51.000000000 +0200
@@ -1611,6 +1611,8 @@
 	widget = shift_to_iframe_parent (widget, &x, &y);
 
 	gdk_window_get_pointer (GTK_LAYOUT (widget)->bin_window, &x, &y, NULL);
+	if (event->is_hint)
+		gdk_device_get_state (event->device, GTK_LAYOUT (widget)->bin_window, NULL, NULL) ;
 
 	/* If there is only a small motion from the origin and simple 
 	   doubleclick is active, abort the motion, otherwise clear simple_dblclick flag */
